### DISCUSSION TITLE
Adding vocabulary support for new words.

### DISCUSSION THREAD

**Discussion by TzurV** [flashlight/flashlight#722] [Q&A] [answered]
I'm running fl_ast_test using the shared models in [RASR release](https://github.com/flashlight/wav2letter/tree/master/recipes/rasr).
I trying to add OOV words by adding them to the lexicon. But they aren't been recognized. 
The words lexicon entry is based on the letters sequence  fl_ast_test is recognizing.

I assume I need to update the LM bin with the new lexicon.
Is the N-Gram used to create the binary [lm_common_crawl_large_4gram](https://dl.fbaipublicfiles.com/wav2letter/rasr/tutorial/lm_common_crawl_large_4gram_prun0-0-5_200kvocab.bin) available ?

**Comment by tlikhomanenko** [comment]
Yep, just change .bin to .arpa and then you can download. I would recommend to train LM on your data where you have new words, then do interpolation between common crawl LM and your LM via SRILM and use it during decoding.

**Reply by TzurV** [reply]
@tlikhomanenko  Thanks. 
What utility/tool should I use to combine the ARPA with the lexicon to create a new bin ?
is it [bin/build_binary text.arpa text.binary](https://kheafield.com/code/kenlm/)  ?

I'm aware of [Types of language models](https://github.com/flashlight/flashlight/blob/master/flashlight/app/asr/README.md#32-types-of-language-models)

**Comment by tlikhomanenko** [comment, marked as answer]
To combine two arpa files to perform interpolation please check docs directly on SRILM tool http://www.speech.sri.com/projects/srilm/ - it will generate a new arpa. Then yes, use kenlm bin/build_binary to do convert arpa into binary.

**Reply by TzurV** [reply]
@tlikhomanenko 
 Thanks for the information. 
How do I mark this as answered ?

**Reply by tlikhomanenko** [reply]
I guess this should be somehow shown to you to choose or google how to do on github, I don't see anything related on my page though.
